src/process.rs

```diff
@@ -64,7 +75,7 @@ pub fn process_id() -> u64 {
 /// counter.send(Inc);
 /// assert_eq!(counter.request(Count), 6);
 /// ```
-pub trait AbstractProcess {
+pub trait AbstractProcess<S = Bincode> {
```

Comment:
Notice how all other traits that take a serializer have the format `<M, S = Bincode>` with a `where S: Serializer<M>` constraint. This is necessary because we are flexible with serializers and don't requirer everyone to support `serde`. For example protobuf has their own `Message` trait and is not dependent on serde.

This means that the `AbstractProcess::Arg` type will only work with serde serializers because of the `type Arg: serde::Serialize + serde::de::DeserializeOwned;` constraint. I would like to avoid this if possible. One solution would be to remove the `type Arg` and move it into a `M` generic. But this means that we need to now carry 2 generic parameter and this overcomplicates everything.